Modules/ExampleApplication/Sources/ExampleApplication/main.swift

```diff
@@ -9,10 +9,28 @@ let port = arguments["port"].int ?? 8080
 let log = LogMiddleware()
 // catches errors and formats them if possible
 let recover = RecoveryMiddleware()
+let session = SessionMiddleware()
 
 // sends requests to different handlers based on their path
 let router = BasicRouter { route in
 
+    route.get("/dash") { request in
+        guard try request.session["loggedIn"].get() ?? false else {
+            return Response(body: "<html><head></head><body>Not logged in. <a href='/login'>login</login></body></html>")
+        }
+        return Response(body: "Hello!")
+    }
+
+    route.get("/login") { request in
+        //TODO: is there anyting we can do about this? request.rawSession.storage mutation 
```

Comment:
use something like this?

https://github.com/Zewo/Zewo/blob/master/Modules/POSIX/Sources/POSIX/Environment.swift#L54